sympy/solvers/tests/test_inequalities.py

```diff
@@ -314,5 +314,10 @@ def test_issue_8974():
     assert isolve(oo > x, x) == And(-oo < x, x < oo)
 
 
+def test_issue_10198():
+    assert reduce_inequalities(-1 + 1/abs( 1/x - 1 ) < 0) == And( -oo < x, x < 0 ) or And( 0 < x, x < 1/2 )
```

Comment:
Please add the following two tests (the 2nd of which indicates why the condition that n be an Integer is present):

```
assert reduce_abs_inequality(-3 + 1/Abs(1 - 1/x), '<', x) == \
    Or(And(-oo < x, x < 0), And(S(0) < x, x < S(3)/4),
    And(S(3)/2 < x, x < oo))
raises(ValueError, 
    lambda: reduce_abs_inequality(-3 + 1/Abs(1 - 1/sqrt(x)), '<', x))
```